src/main/scala/EnsimePlugin.scala

```diff
@@ -400,9 +401,16 @@ object EnsimePlugin extends AutoPlugin {
       artifact = artifactFilter(classifier = Artifact.DocClassifier)
     ).toSet ++ (ensimeUnmanagedJavadocArchives in config in projectRef).gimme
 
-    // needs to take configurations into account
-    // https://github.com/ensime/ensime-sbt/issues/247
-    val deps = project.dependencies.map(_.project.project).map { n => EnsimeProjectId(n, "compile") }
+    val pattern = "([A-Za-z]+)->([A-Za-z]+)".r
+
+    val deps = project.dependencies.map { d =>
+      val name = d.project.project
+      val config = d.configuration.getOrElse("compile") match {
```

Comment:
noob question, isn't it a bad idea call `.get` on a `Option` directly
How about
```scala
val config = d.configuration.map {
  case ivyConfig(from, to) => to
  case _ => "compile"
}.getOrElse("compile")
```